sdk/python/kfp/dsl/_container_op.py

```diff
@@ -847,7 +847,7 @@ def apply(self, mod_func):
               .set_memory_limit('2GB')
           )
         """
-        return mod_func(self)
+        return mod_func(self) or self
```

Comment:
Maybe we should not depend on the mod_func to return anything. 
How about returning the self every time?

Reply:
>Maybe we should not depend on the mod_func to return anything.

This change allows the `mod_func` to return nothing. That's the main idea.

It might still be useful to retain the existing behavior where the function result is used because some functions/decorators might return a different object.

Reply:
Should we enforce the apply function to return the self such that the user will always have a consistent expectation about the returned object. Would you mind explaining a case where the returned object is different while the user really needs the apply to do it?

Reply:
>Would you mind explaining a case where the returned object is different while the user really needs the apply to do it?

It might be useful for things like launchers:

`my_op(arg1, arg2).apply(run_on_cmle(workers=10))`

P.S. I'm not against changing the current behavior. It's just that it wasn't the goal of this PR.

Reply:
Discussed offline. 
Let's keep it this way such that it is backward compatible.